File: flutter/lib/core/crypto/secure_random.dart
import 'dart:math';
import 'dart:typed_data';

/// Cryptographically secure random number generator wrapper
class SecureRandom {
  static final Random _random = Random.secure();

  /// Generate [length] cryptographically secure random bytes
  static Uint8List bytes(int length) {
    final result = Uint8List(length);
    for (int i = 0; i < length; i++) {
      result[i] = _random.nextInt(256);
    }
    return result;
  }

  /// Generate a random integer in range [0, max)
  static int nextInt(int max) {
    return _random.nextInt(max);
  }

  /// Generate a random boolean
  static bool nextBool() {
    return _random.nextBool();
  }
}

File: web/src/core/crypto/secureRandom.ts
/**
 * Cryptographically secure random number generator
 */
export class SecureRandom {
  /**
   * Generate cryptographically secure random bytes
   */
  static bytes(length: number): Uint8Array {
    const array = new Uint8Array(length)
    crypto.getRandomValues(array)
    return array
  }

  /**
   * Generate a random integer in range [0, max)
   */
  static int(max: number): number {
    const array = new Uint32Array(1)
    crypto.getRandomValues(array)
    return array[0] % max
  }

  /**
   * Generate a random boolean
   */
  static bool(): boolean {
    const array = new Uint8Array(1)
    crypto.getRandomValues(array)
    return array[0] % 2 === 0
  }
}
